karabo_data/read_machinery.py

```diff
@@ -183,3 +190,18 @@ def __init__(self, path):
             )
         else:
             self.description = "Unknown data source ({})", datasrc
+
+
+def find_proposal(propno):
+    """Find the proposal directory for a given proposal on Maxwell"""
+    if '/' in propno:
+        # Already passed a proposal directory
+        return propno
+
+    t0 = time.monotonic()
+    for d in iglob(osp.join(DATA_ROOT_DIR, '*/*/{}'.format(propno))):
+        dt = time.monotonic() - t0
+        log.info("Found proposal dir %r in %.2g s", d, dt)
```

Comment:
this is only for debug, right?

Reply:
what's the overhead?

Reply:
The overhead of the logging should be minimal - this is what the logging framework is for, after all. I just timed an unhandled (not printing anything) `log.info()` call at 400 ns.

I'm starting to use logging also in PR #150. We might make the logging visible in some command-line wrappers, and we might ask people to enable it when they run into problems. In this case, I want to be ready to keep an eye on how long it takes to find proposals in case we need to figure out some caching later.

Reply:
Ah sorry, I meant: what is the overhead of scanning into the file system?

Reply:
Good question. I don't know; I'm expecting it to be fairly low at the moment, compared to the cost of opening dozens of HDF5 files once the proposal is found. But it will presumably grow as the number of cycle directories grows.